contracts/meta-oracles/DriftlessHistoricalPriceFeed.sol

```diff
@@ -0,0 +1,277 @@
+ */
+contract DriftlessHistoricalPriceFeed is
+    Ownable,
+    LinkedListLibrary
+{
+    using SafeMath for uint256;
+
+    /* ============ Constants ============ */
+    uint256 constant DAYS_IN_DATASET = 200;
+
+    /* ============ State Variables ============ */
+    uint256 public updateFrequency;
+    uint256 public updateTolerance;
+    uint256 public nextAvailableUpdate;
+    string public dataDescription;
+    IMedian public medianizerInstance;
+
+    LinkedList public historicalPriceData;
+
+    /* ============ Constructor ============ */
+
+    /*
+     * Historical Price Feed constructor.
+     * Stores Historical prices according to passed in oracle address. Updates must be 
+     * triggered off chain to be stored in this smart contract.
+     *
+     * @param  _updateFrequency           How often new data can be logged, passed in seconds
+     * @param  _updateTolerance           If update time exceeds nextAvailable update by this amount
+     *                                    then linearize result, passed in seconds
+     * @param  _medianizerAddress         The oracle address to read historical data from
+     * @param  _dataDescription           Description of data in Data Bank
+     * @param  _seededValues              Array of previous days' Historical price values to seed
+     *                                    initial values in list. Should NOT contain the current
+     *                                    days price.
+     */
+    constructor(
+        uint256 _updateFrequency,
+        uint256 _updateTolerance,
+        address _medianizerAddress,
+        string memory _dataDescription,
+        uint256[] memory _seededValues
+    )
+        public
+    {
+        // Set medianizer address, data description, and instantiate medianizer
+        updateFrequency = _updateFrequency;
+        updateTolerance = _updateTolerance;
+        dataDescription = _dataDescription;
+        medianizerInstance = IMedian(_medianizerAddress);
+
+        // Create initial values array from _seededValues and current price
+        uint256[] memory initialValues = createInitialValues(_seededValues);
+
+        // Define upper data size limit for linked list and input initial value
+        initialize(
+            historicalPriceData,
+            DAYS_IN_DATASET,
+            initialValues[0]
+        );
+
+        // Cycle through input values array (skipping first value used to initialize LinkedList)
+        // and add to historicalPriceData
+        for (uint256 i = 1; i < initialValues.length; i++) {
+            editList(
+                historicalPriceData,
+                initialValues[i]
+            );
+        }
+
+        // Set next available update timestamp
+        nextAvailableUpdate = block.timestamp.add(updateFrequency);
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * Updates linked list with newest data point by querying medianizer. Is eligible to be
```

Comment:
To quickly illustrate how this would work, it may be a worthwhile investment to create a table (like we whiteboarded).